src/common/buffer.cc

```diff
@@ -1491,14 +1821,15 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
     }
   }
 
+  /* reverse previous logic. Now if any buffers can zero-copy then return true */
```

Comment:
1) The reversal was not most precise description that could be used.
The intention is that can_zero_copy returned true if ALL buffers were zero_copy.
Now it returns true if ANY buffer is zero copy.

2) True, there is no performance gain to iterate when all buffers are memory-based. 

3) The iteration for (ptr : buffers) {ptr->copy_to_fd(fd);} is done only when can_zero_copy() is true.